app/src/main/kotlin/com/simplemobiletools/calculator/helpers/MoneyCalculatorImpl.kt

```diff
@@ -96,8 +97,21 @@ class MoneyCalculatorImpl(calculator: Calculator,taxCalculator: TaxCalculator, v
     }
 
     fun performTaxing(location:String){
-        overwriteNumber(TaxOperation(Formatter.stringToDouble(getResult()),location).getResult())
+        overwriteNumber(TaxOperation(getResult(),location).getResult())
     }
 
-    private fun getResult() = mCallback!!.getResult()
-}
+    private fun getResult() = Formatter.stringToDouble(mCallback!!.getResult())
+
+    fun calculateTip(tip: Double) {
+        if (isNumberEmpty()) {
+            overwriteNumber(TipOperation(getResult(), tip).getResult())
+        }
+    }
+
+    private fun isNumberEmpty(): Boolean {
```

Comment:
What is the use of this? Also if the result is bigger than 0, it returns true. Shouldn't it be the opposite?

Reply:
My bad it should be named isNumberValid()

Reply:
👍